### PR TITLE
Implement Initial API Testing Framework for TeamCity

### PR DESCRIPTION
**Pull request by avtikhomirova**
1st iteration

1. Framework Structure: Implemented a comprehensive testing framework covering all essential components.

2. Project Creation Tests: Basic and advanced scenarios including maximum allowed name and ID. Child project creation.
Boundary value analysis.
Negative tests for invalid names, IDs, parent projects, and duplicate projects.

3. Build Configuration Tests: Standard creation process and advanced scenarios with maximum allowed name and ID. Boundary value analysis.
Negative tests for invalid names, IDs, projects, and duplicate build configurations.

### CODE REVIEW

File: .idea/.gitignore
Comment (AlexPshe) [inline .idea/.gitignore:1]: Папку .idea не нужно коммитить, она автоматически создается IDE и хранит IDE настройки 🙌

File: src/test/java/com/example/teamcity/api/BuildConnfigurationTest.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        var testData = projectCreation.getLeft();
+
+        var buildConfDescription = testData.getBuildType();
+        buildConfDescription.setName("");
+
+        new UncheckedBuildConfig(Specifications.getSpec()
+                .authSpec(testData.getUser()))
+                .create(buildConfDescription)
+                .then().assertThat().statusCode(HttpStatus.SC_BAD_REQUEST)
+                .body(Matchers.containsString("When creating a build type, non empty name should be provided"));
```
Comment (AlexPshe) [inline src/test/java/com/example/teamcity/api/BuildConnfigurationTest.java]: Ошибки можно вынести в отдельный Errors статический класс и сделать константами 🙌

File: src/test/java/com/example/teamcity/api/BuildConnfigurationTest.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+        new CheckedBuildConf(Specifications.getSpec()
+                .authSpec(testData.getUser()))
+                .create(testData.getBuildType());
+
+        new UncheckedBuildConfig(Specifications.getSpec()
+                .authSpec(testData.getUser()))
+                .create(testData.getBuildType())
+                .then().assertThat().statusCode(HttpStatus.SC_BAD_REQUEST)
+                .body(Matchers.containsString("The build configuration / template ID \""+ testData.getBuildType().getId() +"\" is already used by another configuration or template"));
```
Comment (AlexPshe) [inline src/test/java/com/example/teamcity/api/BuildConnfigurationTest.java]: Можно реализовать format для более красивой параметризации строки:
https://stackoverflow.com/questions/17364632/java-generating-strings-with-placeholders

**Review by AlexPshe — CHANGES_REQUESTED** [review]
Привет!

Большое спасибо за качественное выполнение задания!

Я добавила несколько комментов по улучшению - нужно пофиксить их за неделю и будет зачтено ⚡️

File: .idea/.gitignore
Comment (avtikhomirova) [reply .idea/.gitignore:1]: Спасибо. добавила в .gitignore

File: src/test/java/com/example/teamcity/api/BuildConnfigurationTest.java (hunk 1 above)
Comment (avtikhomirova) [reply src/test/java/com/example/teamcity/api/BuildConnfigurationTest.java]: Done

File: src/test/java/com/example/teamcity/api/BuildConnfigurationTest.java (hunk 2 above)
Comment (avtikhomirova) [reply src/test/java/com/example/teamcity/api/BuildConnfigurationTest.java]: Done

**Review by AlexPshe — APPROVED** [review]
Отлично, все зачтено!
